doc/source/server_architecture.rst

```diff
@@ -0,0 +1,86 @@
+AIT Server Introduction
+========================
+
+The core of AIT is a "server", :mod:`ait.core.server`, which manages the processing of telemetry, logging, and command messages. The server manages relationships between the following components: 
+
+* **inbound streams**   - inbound streams manage incoming data being recieved by AIT, for example telemtry messages coming into a port.
+* **outbound streams**   - outbound streams manage outgoing data being published by AIT, for example command messages being published to a port.
+* **handlers**   - handlers do the processing of data for inbound and outbound streams. When a message is received on an inbound stream for instance, a handler can handle the processing that needs to be done on the message before it is passed off to outbound streams or plugins.
+* **plugins**       - plugins provide functionality that consume data from inbound streams, do something with it, and provide data to outbound streams. The AIT GUI is a plugin.
+
+The AIT Server uses the messaging library ZeroMQ to manage the passing of data between streams and plugins.
+
+Streams, Plugins and Handlers
```

Comment:
I think adding an individual section on each of the the components makes sense here. Have the main overarching section and then add subsections for Streams, Plugins, and handlers. You could then elaborate on what each of those does in detail, link to the API docs for each, explain creating custom ones, etc. while keeping everything nicely organized.